Anni/wcicd (#16642)
cloudflare/cloudflare-docs: src/content/docs/workers/ci-cd/external-cicd.mdx

---

> pcx_content_type: concept

Reviewer: ```suggestion
pcx_content_type: how-to
```

> title: External CI/CD

Reviewer 2: Rather than "External CI/CD" — which I'm not sure people are going to quite grok intuitively...

Can we have a page at this same level in the hierarchy for Github Actions, and another page for Gitlab Pipelines? For discoverability and SEO?

> description: Integrate Workers development into your existing continuous integration and continuous development workflows.

Reviewer: I'm actually continually surprised at how much keywords in descriptions add for SEO :)

> 1. Go to your CI platform and add the following as secrets:

Reviewer 2: See comment above about separate pages for Github and Gitlab, think it solves for the ambiguity of "Go to your CI platform"

- `CLOUDFLARE_ACCOUNT_ID`: Set to the [Cloudflare account ID](/workers/ci-cd/external-cicd/#cloudflare-account-id) for the account on which you want to deploy your Worker.
- `CLOUDFLARE_API_TOKEN`: Set to the [Cloudflare API token you generated](/workers/ci-cd/external-cicd/#api-token).

:::caution

> It is important not to store the value of `CLOUDFLARE_API_TOKEN` in your repository, as it gives access to deploy Workers on your account. Instead, you should utilize your CI/CD provider's support for storing secrets.

Reviewer 3: ```suggestion
Don't store the value of `CLOUDFLARE_API_TOKEN` in your repository, as it gives access to deploy Workers on your account. Instead, you should utilize your CI/CD provider's support for storing secrets.
```

:::

2. Create a workflow that will be responsible for deploying the Worker. This workflow should run `wrangler deploy`. Review an example [GitHub Actions](https://docs.github.com/en/actions/using-workflows/about-workflows) workflow in the follow section.

### GitHub Actions

> If you use GitHub Actions, Cloudflare provides [an official action](https://github.com/cloudflare/wrangler-action) for deploying Workers. Refer to the following example workflow which deploys your Worker on push to the `main` branch.

Reviewer 2: We should find a way to lead with this piece
